storage/src/vespa/storage/distributor/distributor_bucket_space.cpp

```diff
@@ -118,20 +114,20 @@ DistributorBucketSpace::owns_bucket_in_state(
     return owns_bucket_in_state(*_distribution, clusterState, bucket);
 }
 
-std::vector<uint16_t>
-DistributorBucketSpace::get_ideal_nodes(document::BucketId bucket) const
+const IdealServiceLayerNodesBundle&
+DistributorBucketSpace::get_ideal_service_layer_nodes_bundle(document::BucketId bucket) const
 {
     assert(bucket.getUsedBits() >= _distribution_bits);
-    if (bucket.getUsedBits() > 33) { // cf. storage::lib::Distribution::getStorageSeed
-        // Cannot map to super bucket ==> cannot cache result
-        return get_ideal_nodes_fallback(bucket);
-    }
-    document::BucketId super_bucket(_distribution_bits, bucket.getId());
-    auto itr = _ideal_nodes.find(super_bucket);
+    document::BucketId lookup_bucket((bucket.getUsedBits() > 33) ? bucket.getUsedBits() : _distribution_bits, bucket.getId());
```

Comment:
I tried temporarily adding `assert(bucket.getUsedBits() < 58);`. All unit tests passed. Thus it would have to be a code path not covered by current unit tests.

It is possible to partially disable caching, but should that only be for `bucket.getUsedBits() == 58`, or should it be for `buckt.getUsedBits() > 33` ?

Reply:
As a conservative first step I think we might opt to not cache for > 33 for now. We can then get some time to observe the delta in performance and memory usage for the two cases. We can trivially change it later.